packages/mermaid/src/diagrams/flowchart/flowRenderer.js

```diff
@@ -1,8 +1,9 @@
 import graphlib from 'graphlib';
 import { select, curveLinear, selectAll } from 'd3';
 import { getConfig } from '../../config';
-import dagreD3 from 'dagre-d3';
-import addHtmlLabel from 'dagre-d3/lib/label/add-html-label.js';
+import { render as Render } from 'dagre-d3-es';
+import { applyStyle } from 'dagre-d3-es/src/dagre-js/util';
+import { addHtmlLabel } from 'dagre-d3-es/src/dagre-js/label/add-html-label.js';
```

Comment:
It is normally required in the browser/Node.JS, see https://nodejs.org/api/esm.html#mandatory-file-extensions, but it looks like Vite is allowing it.

My gut feeling is to add the `.js` extension to everything except for things exported directly by the `package.json`. I've changed this in https://github.com/mermaid-js/mermaid/pull/3809/commits/fd76e0e27095997c2ac21902c0629cd6600e30d9 (I've `git rebase`-d my commits).

We could also try to add https://nodejs.org/api/packages.html#subpath-patterns to `dagre-d3-es`, so that we could do stuff like `import {applyStyle} from 'dagre-d3-es/dagre-d3/util.js` (e.g. without the `/src/` bit) instead, but even then, the Node.JS docs seem to recommend only exporting paths that end with `.js` if possible.